ee/clickhouse/queries/funnels/funnel_correlation.py

```diff
@@ -521,17 +538,101 @@ def _run(self) -> Tuple[List[EventOddsRatio], bool]:
         events = positively_correlated_events[:10] + negatively_correlated_events[:10]
         return events, skewed_totals
 
+    def construct_people_url(self, success: bool, event_definition: EventDefinition) -> Optional[str]:
+        #  NOTE: initially only supporting events type, and there is an implicit
+        #  events type when correlation_type is not set.
+        if not self._filter.correlation_type or self._filter.correlation_type == FunnelCorrelationType.EVENTS:
+            # NOTE: we need to convert certain params to strings. I don't think this
+            # class should need to know these details. As we are trying to encode
+            # the filter dict into form encoding, but the persons endpoint is
+            # expecting some attributes to be further encoded as JSON, e.g.
+            # `funnel_correlation_person_entity`, `entities`.
+            # NOTE: I tried using `self._filter.with_data` but this caused errors to
```

Comment:
Interesting: I'm curious, what exactly was the error here?

Reply:
Note that the implementation of with_data is super simple - if it doesn't work, perhaps we should fix it?

We're fundamentally manipulating a filter object here not params so doing it at a params level feels ultimately wrong and something that we'll need to re-do.

Reply:
You could always debug the issue and fix it. Params is not the way forward here.

Reply:
@macobo I have resolve the issue and updated to use `with_data` as well as `to_params` fyi. Is there anything else you'd want me to do here?

Reply:
Ah! Sorry - I was basing my comment off of the last one. Thanks for fixing both! :muscle: